### DISCUSSION TITLE
column must appear in the GROUP BY clause or be used in an aggregate function

### DISCUSSION THREAD

**Discussion by arminus** [apache/superset#18755] [Q&A / Help]
I'm trying to build a simple time series line chart with values from 3 columns:
`|created_on|sensor1|sensor2|sensor3|`
i.e. I'd like to see 3 temperature lines in the chart. 

But I can't seem to figure out how to set my parameters and the error message doesn't really help, since adding all 3 sensors to the group by just results in forever running queries... What am I missing here? I also can't seem to find an example with values in different columns...

![2022-02-16_183911](https://user-images.githubusercontent.com/683680/154325106-358d57fd-026e-4062-880c-c441b4412bcd.png)

**Comment by villebro** [comment]
You should remove `created_on` from your groupby - that column is the implicit x-axis due to being your "Time column" in the "Time" section. Can you try and see what happens.

**Reply by arminus** [reply]
Thanks, but that doesn't change much (I have the 3 sensor columns as 3 custom sql sensor1, sensor2, sensor3 in the metrics definition - just want to see their raw value, no avg, sum or anything)

![2022-02-17_133505](https://user-images.githubusercontent.com/683680/154483002-7b21bb86-f915-4fb7-80dc-3a705b8e97d6.png)

**Comment by arminus** [comment]
Ok, by pure trial&error, I figured out that I have to use a SUM aggregate for each of the 3 sensor columns. Why SUM though? There's no sum being calculated here so why do I need to pick this?

![2022-02-17_142710](https://user-images.githubusercontent.com/683680/154491191-0ba821dd-b96f-487c-83c5-31f778050833.png)

**Comment by MM-Lehmann** [comment]
I am stumbling upon the same weird logic over and over again. I believe it stems from SQL logic itself and there's little Superset could do about it. Although, from a UX perspective, one would wish for `raw` filters instead of `aggregate` filters sometimes. Not sure if it can be done, though
